docs/DeveloperGuide.md

```diff
@@ -148,6 +148,47 @@ The `Storage` component,
+| Email      |                                                | Non-empty string of the format local-part@domain name<br/><ul><li>local-part contains only alphanumeric characters and these special characters, excluding the parantheses, (+_.-)</li><li>local-part cannot start with a special character</li><li>domain name consists of domain labels separated by periods</li><li>domain name must end with domain label at least 2 charaacters long</li><li>each domain label must start and end with alphanumeric characters</li><li>each domain label must consist only of alphanumberic characters, separated only by hyphens (if any)</li></ul> |
+| Remark     | -                                              | Possibly empty string                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     |
+| Tag        | Defaults to empty set when new person is added | Zero or more tag(s), each following the constraint of a Tag                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               |
+
+#### Task
+
+Each Task is implemented by a class with the same name, except for 'Status' which is implemented using a boolean.
+
+Id of each task is unique and automatically generated when task is added. Tasks with the same fields (description, deadline, status, and/or tag) are allowed as a unique Id is generated for each task.
+
+| Field Name  | Description                                                                        | Constraints                                                     |
+|-------------|------------------------------------------------------------------------------------|-----------------------------------------------------------------|
+| Id          | Unique identifier for Task                                                         | Non-empty string                                                |
```

Comment:
Any reason why we have an Id for Task but not Person/Tag? Would be good to include the justification I feel

Reply:
added in latest commit